File: 30_days_C++(modern)_HackerRank_Challenge/AbstractClasses.cpp
#CPP 11 14 20
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Book {
    protected:
        string title;
        string author;
    public:
        Book(string t, string a) {
            title = t;
            author = a;
        }
        virtual void display() = 0;
};


class MyBook : public Book {
    private:
        int price;
    
    public:
        /** * Class Constructor
        * * Parameters:
        * title - The book's title.
        * author - The book's author.
        * price - The book's price.
        **/
        MyBook(string title, string author, int price) 
            : Book(title, author), price(price) {

        }
        

        void display() override {
            cout << "Title: " << title << endl;
            cout << "Author: " << author << endl;
            cout << "Price: " << price << endl;
        }
};


int main() {
    string title, author;
    int price;
    getline(cin, title);
    getline(cin, author);
    cin >> price;
    MyBook novel(title, author, price);
    novel.display();
    return 0;
}

File: 30_days_Python_HackerRank_Challenge/AbstractClasses.py
#python-2
from abc import ABCMeta, abstractmethod
class Book:
    __metaclass__ = ABCMeta
    def __init__(self,title,author):
        self.title=title
        self.author=author   
    @abstractmethod
    def display(): pass

class MyBook(Book):
    def __init__(self, title, author, price):
        Book.__init__(self, title, author)
        self.price = price

    def display(self):
        print "Title:", self.title
        print "Author:", self.author
        print "Price:", self.price

title=raw_input()
author=raw_input()
price=int(raw_input())
new_novel=MyBook(title,author,price)
new_novel.display()

#python-3
from abc import ABCMeta, abstractmethod
class Book(object, metaclass=ABCMeta):
    def __init__(self,title,author):
        self.title=title
        self.author=author   
    @abstractmethod
    def display(): pass

class MyBook(Book):
    def __init__(self, title, author, price):
        super().__init__(title, author)
        self.price = price

    def display(self):
        print("Title:", self.title)
        print("Author:", self.author)
        print("Price:", self.price)

title=input()
author=input()
price=int(input())
new_novel=MyBook(title,author,price)
new_novel.display()
